Lib/ufo2ft/instructionCompiler.py

```diff
@@ -0,0 +1,227 @@
+                    glyf.program._assemble()
+                    glyf.program._disassemble(preserve=True)
+
+            # Handle composites
+            if glyf.isComposite():
+                # Remove empty glyph programs from composite glyphs
+                if hasattr(glyf, "program") and not glyf.program:
+                    delattr(glyf, "program")
+
+                # Set component flags
+
+                # We need to decide when to set the flags automatically.
+                # Let's assume if any lib key is not there, or the component
+                # doesn't have an identifier, we should autoset all component
+                # flags.
+                for i, c in enumerate(glyf.components):
+                    if i >= len(glyph.components):
+                        logger.error(
+                            "Number of components differ between UFO and TTF "
+                            f"in glyph '{name}' ({len(glyph.components)} vs. "
+                            f"{len(glyf.components)}, not setting flags in "
+                            "additional components."
+                        )
+                        break
+                    ufo_component = glyph.components[i]
+                    if (
+                        ufo_component.identifier is None
+                        or "public.objectLibs" not in glyph.lib
+                        or "public.objectIdentifiers"
+                        not in glyph.lib["public.objectLibs"]
+                        or ufo_component.identifier
+                        not in glyph.lib["public.objectLibs"][
+                            "public.objectIdentifiers"
+                        ]
+                        or "public.truetype.instructions"
+                        not in glyph.lib["public.objectLibs"][
+                            "public.objectIdentifiers"
+                        ][ufo_component.identifier]
+                    ):
+                        # Auto set
+
+                        # We don't try to set the "OVERLAP_COMPOUND" flag
+
+                        # Set "ROUND_XY_TO_GRID" if the component has an offset
+                        # c.flags &= ~ROUND_XY_TO_GRID
+                        # if c.x != 0 or c.y != 0:
+                        #     c.flags |= ROUND_XY_TO_GRID
+
+                        # Nope, don't change the flag:
+                        # https://github.com/googlefonts/ufo2ft/pull/425
+                        pass
+
+                        # "USE_MY_METRICS" has been set already by
+                        # outlineCompiler.OutlineTTFCompiler.autoUseMyMetrics
+
+                    else:
+                        # Use values from lib
+
+                        flags = glyph.lib["public.objectLibs"][
+                            "public.objectIdentifiers"
+                        ][glyph.components[i].identifier][
+                            "public.truetype.instructions"
+                        ]
```

Comment:
This should use `glib.lib["public.objectLibs"][glyph.components[i].identifier]` directly.